pallets/anchor-handler/src/tests_signature_bridge.rs

```diff
@@ -0,0 +1,457 @@
+			));
+
+			assert_ok!(SignatureBridge::execute_proposal(
+				Origin::signed(RELAYER_A),
+				prop_id,
+				src_id,
+				r_id,
+				Box::new(proposal.clone()),
+				sig.0.to_vec(),
+			));
+			assert_eq!(1, Counts::<Test>::get(src_id));
+
+			// the anchor-handler callback must have been called by bridge
+			// event must be emitted in callback should exist
+			event_exists(crate::Event::AnchorEdgeAdded);
+			// edge count should be 1
+			assert_eq!(
+				1,
+				<pallet_linkable_tree::EdgeList<Test>>::iter_prefix_values(0)
+					.into_iter()
+					.count()
+			);
+
+			let expected_tree_id = 0;
+			assert_eq!(
+				edge_metadata,
+				<pallet_linkable_tree::EdgeList<Test>>::get(expected_tree_id, src_id)
+			);
+
+			let expected_update_record =
+				UpdateRecord { tree_id: expected_tree_id, resource_id: r_id, edge_metadata };
+			assert_eq!(expected_update_record, UpdateRecords::<Test>::get(src_id, 0));
+
+			// Update Edge
+			let root = Element::from_bytes(&[2; 32]);
+			let latest_leaf_index = 10;
+			let edge_metadata = EdgeMetadata { src_chain_id: src_id, root, latest_leaf_index };
+
+			let proposal = make_anchor_update_proposal(&r_id, edge_metadata.clone());
+			let msg = keccak_256(&proposal.encode());
+			let sig: Signature = pair.sign_prehashed(&msg).into();
+
+			assert_ok!(SignatureBridge::execute_proposal(
+				Origin::signed(RELAYER_A),
+				prop_id + 1,
+				src_id,
+				r_id,
+				Box::new(proposal.clone()),
+				sig.0.to_vec(),
+			));
+
+			assert_eq!(2, Counts::<Test>::get(src_id));
+
+			// the anchor-handler callback must have been called by bridge
+			// event must be emitted in callback should exist
+			event_exists(crate::Event::AnchorEdgeUpdated);
+			// edge count should be 1
+			assert_eq!(
+				1,
+				<pallet_linkable_tree::EdgeList<Test>>::iter_prefix_values(0)
+					.into_iter()
+					.count()
+			);
+			assert_eq!(
+				edge_metadata,
+				<pallet_linkable_tree::EdgeList<Test>>::get(expected_tree_id, src_id)
+			);
+			let expected_update_record =
+				UpdateRecord { tree_id: expected_tree_id, resource_id: r_id, edge_metadata };
+			assert_eq!(expected_update_record, UpdateRecords::<Test>::get(src_id, 1));
+		})
+}
+
+#[test]
+fn should_fail_with_chain_already_whitelisted() {
+	let src_id = 1u32;
+	let r_id = derive_resource_id(src_id, b"execute_anchor_create_proposal");
+	let public_uncompressed = hex!("8db55b05db86c0b1786ca49f095d76344c9e6056b2f02701a7e7f3c20aabfd913ebbe148dd17c56551a52952371071a6c604b3f3abe8f2c8fa742158ea6dd7d4");
+	let pair = ecdsa::Pair::from_string(
+		"0x9d61b19deffd5a60ba844af492ec2cc44449c5697b326919703bac031cae7f60",
+		None,
+	)
+	.unwrap();
+
+	new_test_ext_initialized(src_id, r_id, b"AnchorHandler.execute_anchor_create_proposal".to_vec())
+		.execute_with(|| {
+			assert_err!(
+				SignatureBridge::whitelist_chain(Origin::root(), src_id),
+				pallet_signature_bridge::Error::<Test, _>::ChainAlreadyWhitelisted
+			);
+		})
+}
+
+#[test]
+fn should_fail_with_invalid_chain_id() {
```

Comment:
What should fail? Does this fail because it's whitelisting "this" chain's ID as a whitelisted chain or is there another reason this is failing?